docs/versioned_docs/version-v3.2.0/02-Features/endpoint-validation.md

```diff
@@ -0,0 +1,215 @@
+---
+
+# Endpoint validation
+
+We provide support for endpoint validation, when implementing your own custom web hook.
+
+- [Endpoint validation](#endpoint-validation)
+  - [Installation](#installation)
+  - [Azure Event Grid authorization](#azure-event-grid-authorization)
+    - [Enforce authorization globally](#enforce-authorization-globally)
+      - [Configuration](#configuration)
+    - [Enforce authorization per controller or operation](#enforce-authorization-per-controller-or-operation)
+      - [Configuration](#configuration-1)
+  - [Azure Event Grid subscription validation](#azure-event-grid-subscription-validation)
+    - [Enforce subscription validation per controller or operation](#enforce-subscription-validation-per-controller-or-operation)
+    - [Use subscription validation in Azure Functions](#use-subscription-validation-in-azure-functions)
+
+## Installation
+
+The features described here require the following package:
+
+> **⚠ This package used to be called `Arcus.EventGrid.Security`. Please make sure that you migrate towards this package as it's not being actively maintained beyond v3.2.**
+
+```shell
+PM> Install-Package Arcus.EventGrid.Security.WebApi
+```
+## Azure Event Grid authorization
+
+Azure Event Grid authorization allows to secure your webhook with a secret key (taken from the query string or an HTTP header). 
+This is needed, because Azure Event Grid is send a validation request to a newly configured web hook, in order to prevent people leveraging Azure Event Grid to bring down a 3rd party API. 
+
+The implementation we provide, is echoing back the validation key on your operation, in order to have the validation by Event Grid out of the box.
+
+### Enforce authorization globally
+
+We created the `EventGridAuthorizationFilter` MVC filter that will secure the endpoint and handle the handshake.
+
+```csharp
+using Arcus.EventGrid.WebApi.Security;
+using Microsoft.Extensions.DependencyInjection;
+
+public class Startup
+{
+    public void ConfigureService(IServiceCollection services)
+    {
+        // Looks for the 'x-api-key' header in the HTTP request and tries to match it with the secret retrieved in the secret store with the name 'MySecret'.
+        services.AddMvc(options => options.Filters.AddEventGridAuthorization(HttpRequestProperty.Header, "x-api-key", "MySecret")));
+    }
+}
+```
+
+For this setup to work, an Arcus secret store is required as the provided secret name (in this case `"MySecret"`) will be looked up.
+See [our offical documentation](https://security.arcus-azure.net/features/secret-store/) for more information about setting this up.
+
+#### Configuration
+
+The `EventGridAuthorizationFilter` has some additional consumer-configurable options to influence the behavior of the authorization.
+
+```csharp
+using Arcus.EventGrid.WebApi.Security;
+using Microsoft.Extensions.DependencyInjection;
+
+public class Startup
+{
+    public void ConfigureService(IServiceCollection services)
```

Comment:
Isn't this ConfigureServices ?

Reply:
Yes, that's right 😅 , will change that. Thx!

Reply:
Done in https://github.com/arcus-azure/arcus.eventgrid/pull/232/commits/bbc1e0949e9829439e511434056f18480cdffe1c